opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1228,18 +1204,21 @@ namespace Opm {
 
 
     template<typename TypeTag>
-    std::pair<bool, bool>
+    bool
     BlackoilWellModel<TypeTag>::
     updateWellControlsAndNetworkIteration(const bool mandatory_network_balance,
-                                          const bool relax_network_tolerance,
+                                          const int network_update_iteration,
                                           const double dt,
-                                          DeferredLogger& local_deferredLogger)
+                                          bool& well_group_control_changed,
+                                          DeferredLogger& deferred_logger)
     {
         OPM_TIMEFUNCTION();
-        auto [well_group_control_changed, more_inner_network_update] =
-                updateWellControls(mandatory_network_balance,
-                                   local_deferredLogger,
-                                   relax_network_tolerance);
+        const int iterationIdx = simulator_.model().newtonMethod().numIterations();
+        const int episodeIdx = simulator_.episodeIndex();
+        const auto& comm = simulator_.vanguard().grid().comm();
+        this->updateAndCommunicateGroupData(episodeIdx, iterationIdx, param_.nupcol_group_rate_tolerance_, deferred_logger);
```

Comment:
In the original code, this communication is done after
```c++
       if (!this->wellsActive() && !network.active()) {
            return {false, false};
        }
```

Do you think it is potentially introducing some extra communications?